frontend/types/hashTabs.ts

```diff
@@ -1,5 +1,12 @@
+import type { IconDefinition } from '@fortawesome/fontawesome-svg-core'
+import type { Component } from 'vue'
+
 export type HashTab = {
+  component?: Component,
   disabled?: boolean,
-  index: number,
+  icon?: IconDefinition,
+  index: string,
+  placeholder?: string,
+  title?: string,
 }
 export type HashTabs = Record<string, HashTab>
```

Comment:
couldn't that be an array instead of an object, then we would spare us the struggle of sorting this

Reply:
originally it was easier with a Record to map the url hash ... but now that the PrimeVue tabs no longer use an index but a string as key it's much easier and cleaner with an array.